gap/pkg/OscarInterface/gap/QQBar.gi

```diff
@@ -0,0 +1,195 @@
+  { x, y } -> QQBarFieldElement( x ) = y );
+
+
+InstallMethod( \<,
+   "for QQBarField elements",
+ [ "IsQQBarFieldElement", "IsQQBarFieldElement" ],
+ { x, y } -> JuliaPointer( x ) < JuliaPointer( y ) );
+
+InstallMethod( \<,
+ [ "IsQQBarFieldElement", "IsCyc" ],
+ { x, y } -> x < QQBarFieldElement( y ) );
+
+InstallMethod( \<,
+ [ "IsCyc", "IsQQBarFieldElement" ],
+ { x, y } -> QQBarFieldElement( x ) < y );
+
+InstallMethod( \in,
+  [ "IsObject", "IsQQBarField" ], SUM_FLAGS,
+  { x, F } -> IsQQBarFieldElement( x ) );
+
+
+InstallMethod( \+,
+  [ "IsQQBarFieldElement", "IsQQBarFieldElement" ],
+  { x, y } -> QQBarFieldElement( JuliaPointer( x ) + JuliaPointer( y ) ) );
+
+InstallMethod( \+,
+  [ "IsCyclotomic", "IsQQBarFieldElement" ],
+  { x, y } -> QQBarFieldElement( x ) + y );
+
+InstallMethod( \+,
+  [ "IsQQBarFieldElement", "IsCyclotomic" ],
+  { x, y } -> x + QQBarFieldElement( y ) );
+
+
+InstallMethod( AdditiveInverseSameMutability,
+  [ "IsQQBarFieldElement" ],
+  x -> QQBarFieldElement( - JuliaPointer( x ) ) );
+
+InstallMethod( AdditiveInverseMutable,
+  [ "IsQQBarFieldElement" ],
+  x -> QQBarFieldElement( - JuliaPointer( x ) ) );
+
+InstallMethod( \*,
+  [ "IsQQBarFieldElement", "IsQQBarFieldElement" ],
+  { x, y } -> QQBarFieldElement( JuliaPointer( x ) * JuliaPointer( y ) ) );
+
+InstallMethod( \*,
+  [ "IsQQBarFieldElement", "IsCyc" ],
+  { x, y } -> x * QQBarFieldElement( y ) );
+
+InstallMethod( \*,
+  [ "IsCyc", "IsQQBarFieldElement" ],
+  { x, y } -> QQBarFieldElement( x ) * y );
+
+InstallMethod( InverseOp,
+  [ "IsQQBarFieldElement" ],
+  x -> QQBarFieldElement( Oscar.inv( JuliaPointer( x ) ) ) );
+
+InstallMethod( ComplexConjugate,
+  [ "IsQQBarFieldElement" ],
+  x -> QQBarFieldElement( Julia.conj( JuliaPointer( x ) ) ) );
+
+InstallMethod( Sqrt,
+  [ "IsQQBarFieldElement" ],
+  function( x )
+  return QQBarFieldElement( Julia.sqrt( x ) );
+  end );
+
+InstallMethod( DefaultFieldOfMatrix,
+  [ "IsMatrix and IsQQBarFieldElementCollColl" ],
+  M -> QQBarField );
+
+InstallMethod( DefaultFieldByGenerators,
+  [ "IsList and IsQQBarFieldElementCollection" ],
+  L -> QQBarField );
+
+# The `rand` method for `QQBarFieldElem` in Oscar is based on
+# `_flint_rand_states`, we cannot use a GAP random source for it,
+# thus `InstallMethodWithRandomSource` cannot be used.
+InstallMethod( Random,
+  [ "IsQQBarField" ],
+  F -> QQBarFieldElement( CallJuliaFunctionWithKeywordArguments( Oscar.rand,
+                            [ QQBarField_Julia ],
+                            rec( degree:= 4, bits:= 5 ) ) ) );
+
+InstallOtherMethod( AbsoluteValue,
+  [ "IsQQBarFieldElement" ], SUM_FLAGS,
+  x -> QQBarFieldElement( Julia.abs( JuliaPointer( x ) ) ) );
+
+InstallMethod( Eigenvalues,
+  [ "IsQQBarField", "IsMatrix and IsQQBarFieldElementCollColl" ],
+  function( F, M )
+  local MM;
+
+  if QQBarFieldElementMatrixType = fail then
+    QQBarFieldElementMatrixType:= JuliaEvalString( "Matrix{QQBarFieldElem}" );
```

Comment:
Thinking some more, this won't work if the user did not do `using Oscar` (so e.g. if Oscar is only loaded indirectly)...